tests/test_guider.py

```diff
@@ -0,0 +1,189 @@
+# This file is part of summit_utils.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (https://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+
+import os
+import tempfile
+import unittest
+from functools import wraps
+
+import numpy as np
+import pandas as pd
+
+import lsst.utils.tests
+from lsst.daf.butler import Butler
+from lsst.summit.utils.butlerUtils import makeDefaultButler
+from lsst.summit.utils.guiders.detection import GuiderStarTracker
+from lsst.summit.utils.guiders.plotting import GuiderPlotter
+from lsst.summit.utils.guiders.reading import GuiderReader
+from lsst.summit.utils.utils import getSite
+
+
+def check_plot_file_size(size_threshold=1000):
+    """Decorator factory to check that a plot file is created
+    and is of sufficient size (in bytes)."""
+
+    def decorator(func):
```

Comment:
We could have a quick chat about why I don't like this pattern, but in short: I'd just use a normal context manager around the plots and then check the file sizes inside that context - basically, this is a bit too magic, and not necessary.

Reply:
Removed.